### PR TITLE
Fix: Do not remove space after comma in string value

### LINKED ISSUE #2: \Zend\Json\Json::prettyPrint fails for json containing string with commas
Having the following JSON:

``` json
{ "key" : "Hello, world!" }
```

and parsing it using

``` php
\Zend\Json\Json::prettyPrint('{ "key" : "Hello, world!" }', ["indent" => ""]);
```

will produce:

``` php
{"key":"Hello,world!"}
```

The reason is first line from \Zend\Json\Json::prettyPrint

``` php
$tokens = preg_split('|([\{\}\]\[,])|', $json, -1, PREG_SPLIT_DELIM_CAPTURE);
```

not taking into account that a comma could resides inside a token.

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] asserts that a space after a `,` in `string` value is left alone
* [x] leaves the space after a `,` in a string value alone

Fixes #2.

💁‍♂️ Not sure if a fix yet, but figured I provide a failing test at least.

### CODE REVIEW

**Comment by weierophinney** [comment]
@localheinz I've rebased it off of current master, and the tests demonstrate the issue.

I've no idea how to fix it presently; carry on if you want, or ping me or others for assistance if you need to.

**Comment by michalbundyra** [comment]
@localheinz @weierophinney 
I've resolved the issue. First I've removed duplicated tests - somehow after @weierophinney rebase some tests were added twice. Then I've changed completely approach in `prettyPrint` so all tests are now passing. Added more tests to check the behaviour for other edge cases.

Let me know what you think, and if you ok with it, I'll include it in the next minor release. Thanks!

File: src/Json.php [hunk 1]
```diff
@@ -165,84 +165,107 @@ protected static function recursiveJsonExprFinder(
...
+                        $result .= $json[$i];
+                        while (isset($json[$i + 1]) && preg_match('/\s/', $json[$i + 1])) {
+                            ++$i;
+                        }
+                        if (isset($json[$i + 1]) && $json[$i + 1] !== '}' && $json[$i + 1] !== ']') {
+                            $result .= "\n" . str_repeat($indentString, count($stack));
+                        }
+                        continue 2;
+                    }
+                    break;
```
Comment (weierophinney) [inline src/Json.php]: I'd break early, so that you can remove a level of indentation:

```php
if ($inLiteral) {
    break;
}

$stack[] = $json[$i];

$result .= $json[$i];
while (isset($json[$i + 1]) && preg_match('/\s/', $json[$i+1])) {
    ++$i;
}
if (isset($json[$i + 1]) && $json[$i + 1] !== '}' && $json[$i + 1] !== ']') {
    $result .= "\n" . str_repeat($indentString, count($stack));
}
continue 2;
```

I'd do that in each of the case statements below as well.

File: src/Json.php [hunk 2]
```diff
@@ -165,84 +165,107 @@ protected static function recursiveJsonExprFinder(
...
-                $indentLevel++;
-                if ($result != '' && $result[strlen($result) - 1] === "\n") {
-                    $result .= $prefix;
-                }
-                $result .= $token;
-
-                $closingBracket = $bracketPairs[$token];
-
-                do {
+            if (! $inLiteral) {
```
Comment (weierophinney) [inline src/Json.php]: Here, I would return early:

```php
if ($inLiteral) {
    return;
}
// promote the conditional expression up a level here
```

**Review by weierophinney — CHANGES_REQUESTED** [review]
Solution looks good - I have some suggestions for readability below.

File: src/Json.php (hunk 2 above)
Comment (michalbundyra) [reply src/Json.php]: I can't as this is inside `for` loop.

File: src/Json.php (hunk 1 above)
Comment (michalbundyra) [reply src/Json.php]: Updated, thanks!

File: src/Json.php (hunk 2 above)
Comment (weierophinney) [reply src/Json.php]: So do a `continue` instead of a `return`. :)

Comment (michalbundyra) [reply src/Json.php]: Updated, but honestly I am not sure if it is clearer.

Comment (weierophinney) [reply src/Json.php]: I just viewed the file as committed (i.e., not the diff, but the result following diff), and it is much clearer to me that if we are in a literal and have not hit a character that would mark the end of the literal, we advance to the next character and do nothing. So definitely a win!

**Review by weierophinney — APPROVED** [review]
:ship:

**Comment by michalbundyra** [comment]
Thanks, @localheinz!
